src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ResettableValueTaskSource.cs

```diff
@@ -310,7 +310,15 @@ public Task GetTask(object? keepAlive)
         {
             if (_finalTaskSource is null)
             {
+                if (_isSignaled)
```

Comment:
```suggestion
                if (_isCompleted && _isSignaled)
```
`TrySignal` would return if it's not completed, you're bypassing that logic here.

Reply:
`_isSignaled` being set implies `_isCompleted`:
https://github.com/dotnet/runtime/blob/2d0dff44ea5c6d3c7d42b80b77bf8abc90072049/src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/ResettableValueTaskSource.cs#L342-L348